grpc-protocol/src/main/java/com/linecorp/armeria/client/grpc/protocol/UnaryGrpcClient.java

```diff
@@ -162,27 +188,62 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) {
 
                            final ArmeriaMessageDeframer deframer =
                                    new ArmeriaMessageDeframer(Integer.MAX_VALUE);
-                           msg.toHttpResponse().decode(deframer, ctx.alloc())
-                              .subscribe(singleSubscriber(msg, responseFuture), ctx.eventLoop());
+                           msg.toHttpResponse().decode(deframer, ctx.alloc(),
+                                                       byteBufConverter(ctx.alloc(), isGrpcWebText))
+                              .subscribe(singleSubscriber(ctx, msg, serializationFormat, responseFuture),
+                                         ctx.eventLoop());
                            return responseFuture;
                        }), ctx.eventLoop());
         }
 
         private static Subscriber<DeframedMessage> singleSubscriber(
-                AggregatedHttpResponse msg, CompletableFuture<HttpResponse> responseFuture) {
+                ClientRequestContext ctx, AggregatedHttpResponse msg, SerializationFormat serializationFormat,
+                CompletableFuture<HttpResponse> responseFuture) {
 
             return new Subscriber<DeframedMessage>() {
+                private HttpData content = HttpData.empty();
+                @Nullable
+                private HttpHeaders trailers;
+
                 @Override
                 public void onSubscribe(Subscription s) {
-                    s.request(1);
+                    // At least 2 requests are required for receiving trailers.
+                    s.request(2);
+                }
+
+                private boolean isTrailers(DeframedMessage message) {
+                   return message.type() >> 7 == 1;
                 }
 
                 @Override
-                public void onNext(DeframedMessage unframed) {
-                    final ByteBuf buf = unframed.buf();
+                public void onNext(DeframedMessage message) {
+                    if (UnaryGrpcSerializationFormats.isGrpcWeb(serializationFormat) && isTrailers(message)) {
+                        final ByteBuf buf;
+                        try {
+                            buf = InternalGrpcWebUtil.messageBuf(message, ctx.alloc());
+                        } catch (Throwable t) {
+                            responseFuture.completeExceptionally(t);
```

Comment:
How about releasing `content` when an exception is raised?

Reply:
Are you referring to `buf`? In this scope it's not going to be initialized, and inside `InternalGrpcWebUtil.messageBuf` it seems to be released if the exception is raised. Am I missing something or you're referring to something else?

Reply:
I figured it out, PTAL.